### PR TITLE
Fix res import errors

### PR DESCRIPTION
**Pull request by eivindjahren**
**Issue**

#2503

**Approach**
* Adds `__all__` to `__init__`
* Import implicit dependencies in the module it is used
* Use typing to make the use of the import explicit
* Remove unused dependencies

Several files would rely on importing from the `__init__.py` of its module. This is quite fragile as the order of imports in the `__init__.py` must facilitate a partial instantiation in order to avoid circular imports. As more imports are added to resolve the above issue, it becomes very difficult to continue with that scheme. Therefore imports such as `from . import EnkFMain` is changed to `from .enkf_main import EnkFMain` for any file inside `res/enkf` while files outside `res/enkf` can use `from res.enkf import EnkFMain`.

### CODE REVIEW

File: res/sched/history.py [hunk 1]
```diff
@@ -31,7 +34,7 @@ class History(BaseCClass):
     )
     _free = ResPrototype("void  history_free( history )")
 
-    def __init__(self, refcase=None, use_history=False, sched_file=None):
+    def __init__(self, refcase=Optional[EclSum], use_history=False, sched_file=None):
```
Comment (dotfloat) [inline res/sched/history.py:37]:
```suggestion
    def __init__(self, refcase: Optional[EclSum] = None, use_history=False, sched_file=None):
```

File: res/enkf/plot_data/ensemble_plot_gen_kw.py [hunk 2]
```diff
@@ -54,7 +56,7 @@ def __init__(self, ensemble_config_node, file_system, input_mask=None):
 
         self.__load(file_system, input_mask)
 
-    def __load(self, file_system, input_mask=None):
+    def __load(self, file_system: EnkfFs, input_mask: BoolVector = None):
```
Comment (dotfloat) [inline res/enkf/plot_data/ensemble_plot_gen_kw.py:59]:
```suggestion
    def __load(self, file_system: EnkfFs, input_mask: Optional[BoolVector] = None):
```

File: res/enkf/enkf_state.py [hunk 3]
```diff
@@ -45,7 +50,10 @@ def ensembleConfig(self):
         return self._get_ens_config()
 
     def initialize(
-        self, fs, param_list=None, init_mode=EnkfInitModeEnum.INIT_CONDITIONAL
+        self,
+        fs: EnkfFs,
+        param_list: StringList = None,
```
Comment (dotfloat) [inline res/enkf/enkf_state.py:55]:
```suggestion
        param_list: Optional[StringList] = None,
```

File: res/enkf/analysis_config.py [hunk 4]
```diff
@@ -104,7 +103,12 @@ class AnalysisConfig(BaseCClass):
         "int analysis_config_get_min_realisations(analysis_config)"
     )
 
-    def __init__(self, user_config_file=None, config_content=None, config_dict=None):
+    def __init__(
+        self,
+        user_config_file=None,
+        config_content: ConfigContent = None,
```
Comment (dotfloat) [inline res/enkf/analysis_config.py:109]:
```suggestion
        config_content: Optional[ConfigContent] = None,
```

**Review by dotfloat — CHANGES_REQUESTED** [review]

File: res/enkf/analysis_config.py (hunk 4 above)
Comment (eivindjahren) [reply res/enkf/analysis_config.py]: Fixed now

File: res/enkf/enkf_state.py (hunk 3 above)
Comment (eivindjahren) [reply res/enkf/enkf_state.py]: Fixed now

File: res/enkf/plot_data/ensemble_plot_gen_kw.py (hunk 2 above)
Comment (eivindjahren) [reply res/enkf/plot_data/ensemble_plot_gen_kw.py]: Fixed now

File: res/sched/history.py (hunk 1 above)
Comment (eivindjahren) [reply res/sched/history.py]: Oups, nice catch. Fixed

**Review by dotfloat — APPROVED** [review]

**Comment by eivindjahren** [comment]
Jenkins test this please

**Pull request merged**